include/mata/nfa-algorithms.hh

```diff
@@ -38,102 +36,100 @@
+ * @param[in] bigger Automaton which language should include the smaller one
+ * @param[in] alphabet Alphabet of both automata (not needed for antichain algorithm)
+ * @param[out] cex A potential counterexample word which breaks inclusion
+ * @return True if smaller language is included,
+ * i.e., if the final intersection of smaller complement of bigger is empty.
+ */
+bool is_included_antichains(const Nfa& smaller, const Nfa& bigger, const Alphabet*  alphabet = nullptr, Run* cex = nullptr);
 
-    /**
-     * Universality checking based on subset construction with antichain.
-     * @param[in] aut Automaton which universality is checked
-     * @param[in] alphabet Alphabet of the automaton
-     * @param[out] cex Counterexample word which eventually breaks the universality
-     * @return True if the automaton is universal, otherwise false.
-     */
-    bool is_universal_antichains(
-            const Nfa&         aut,
-            const Alphabet&    alphabet,
-            Run*              cex);
+/**
+ * Universality check implemented by checking emptiness of complemented automaton
+ * @param[in] aut Automaton which universality is checked
+ * @param[in] alphabet Alphabet of the automaton
+ * @param[out] cex Counterexample word which eventually breaks the universality
+ * @return True if the complemented automaton has non empty language, i.e., the original one is not universal
+ */
+bool is_universal_naive(const Nfa& aut, const Alphabet& alphabet, Run* cex);
 
-    Simlib::Util::BinaryRelation compute_relation(
-            const Nfa& aut,
-            const StringMap&  params = {{"relation", "simulation"}, {"direction", "forward"}});
+/**
+ * Universality checking based on subset construction with antichain.
+ * @param[in] aut Automaton which universality is checked
+ * @param[in] alphabet Alphabet of the automaton
+ * @param[out] cex Counterexample word which eventually breaks the universality
+ * @return True if the automaton is universal, otherwise false.
+ */
+bool is_universal_antichains(const Nfa& aut, const Alphabet& alphabet, Run* cex);
+
+Simlib::Util::BinaryRelation compute_relation(
+        const Nfa& aut,
+        const StringMap&  params = {{"relation", "simulation"}, {"direction", "forward"}});
+
+/**
+ * @brief Compute intersection of two NFAs with a possibility of using multiple epsilons.
+ *
+ * @param[in] lhs First NFA to compute intersection for.
+ * @param[in] rhs Second NFA to compute intersection for.
+ * @param[in] preserve_epsilon Whether to compute intersection preserving epsilon transitions.
+ * @param[in] epsilons Set of symbols to be considered as epsilons
+ * @param[out] prod_map Mapping of pairs of the original states (lhs_state, rhs_state) to new product states.
+ * @return NFA as a product of NFAs @p lhs and @p rhs with ε-transitions preserved.
+ */
+Nfa intersection_eps(const Nfa& lhs, const Nfa& rhs, bool preserve_epsilon, const std::set<Symbol>& epsilons,
```

Comment:
Suggestion: Rename to `intersection_with_eps`?

Reply:
That can be done, but not for this PR. I do not want any more changes which actually change anything in the interface. Will add this to my todo list.

Reply:
Yes, this also should be discussed with others (probably).

Reply:
I believe that the function was already named as you suggest, but then some objected that the name is too long.